### PR TITLE
mangleToBuffer: use existing deco rather than regenerate it

### PR DESCRIPTION
**Pull request by WalterBright**
Should make things a tad faster.

### CODE REVIEW

File: src/dmangle.d [hunk 1]
```diff
@@ -833,8 +833,13 @@ extern (C++) const(char)* mangleExact(FuncDeclaration fd)
 
 extern (C++) void mangleToBuffer(Type t, OutBuffer* buf)
 {
-    scope Mangler v = new Mangler(buf);
-    v.visitWithMask(t, 0);
+    if (t.deco)
+        buf.writestring(t.deco);
```
Comment (ZombineDev) [inline src/dmangle.d]: `visitWithMask` already contains this caching logic. See:
https://github.com/dlang/dmd/blob/f2dc24d56f347c6640147aa14b0b6e1e278f667b/src/dmangle.d#L178

Comment (WalterBright) [inline src/dmangle.d]: You are correct, but there'd still be the setup/teardown of Mangler.

**Comment by andralex** [comment]
@ZombineDev leaving the review to you

File: src/dmangle.d (hunk 1 above)
Comment (rainers) [inline src/dmangle.d]: AFAICT the only place where this overload is called is `Type.merge` in case deco is null. So the added case will not be triggered.

Comment (WalterBright) [inline src/dmangle.d]: It's also called in objc.d, and I intend to replace the mangleToBuffer() in mtype.d:getTypeInfoIdent() with a call to it. (The `internal` int is always 0.)

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**